karabo_data/export.py

```diff
@@ -188,20 +188,33 @@ def feed(self, data, metadata=None):
         self._buffer.put(self._serialize(data, metadata))
 
 
-def main(argv=None):
-    ap = ArgumentParser(prog="karabo-bridge-serve-files")
-    ap.add_argument("path", help="Path of a file or run directory to serve")
-    ap.add_argument("port", help="TCP port to run server on")
-    args = ap.parse_args(argv)
+def serve_files(path, port, **kwargs):
+    """Stream data from files through a TCP socket.
 
-    if osp.isdir(args.path):
-        data = RunDirectory(args.path)
+    Parameters
+    ----------
+    path: str
+        Path to the HDF5 file or file folder.
+    port: int
+        Local TCP port to bind socket to.
+    """
+    if osp.isdir(path):
+        data = RunDirectory(path)
     else:
-        data = H5File(args.path)
+        data = H5File(path)
 
-    streamer = ZMQStreamer(args.port)
+    streamer = ZMQStreamer(port, **kwargs)
     streamer.start()
     for tid, train_data in data.trains():
-        streamer.feed(train_data)
-
+        if train_data:
```

Comment:
Have you come across some case where this `if` isn't satisfied?

Reply:
Yes, the train_data could be empty. I found that in the run that Dmitry used for test - r0078. The code crashed with the empty train_data:

  File "/home/zhujun/karabo_data/karabo_data/export.py", line 45, in run
    interface.send_multipart(self.buffer.get())
  File "/usr/local/lib/python3.6/dist-packages/zmq/sugar/socket.py", line 438, in send_multipart
    return self.send(msg_parts[-1], flags, copy=copy, track=track)
IndexError: list index out of range